Phonebook.Backend/Phonebook.Backend.PictureService/Phonebook.Backend.PictureService/Controllers/EmployeePicturesController.cs

```diff
@@ -0,0 +1,149 @@
+        {
+            if (file == null || file.Length == 0)
+            {
+                return BadRequest("No Payload.");
+            }
+            var user = HttpContext.User;
+            if (user.Identity.Name.Split("\\")[1] != id)
+            {
+                return StatusCode(401, "User not authorized.");
+            }
+            if (
+                // BMP
+                file.ContentType == "image/bmp" ||
+                file.ContentType == "image/x-ms-bmp" ||
+                file.ContentType == "image/x-bmp" ||
+                // SVG
+                file.ContentType == "image/svg+xml" ||
+                // WebP
+                file.ContentType == "image/webp" ||
+                // GIF
+                file.ContentType == "image/gif" ||
+                // JPEG 2000
+                file.ContentType == "image/jp2" ||
+                file.ContentType == "image/jpx" ||
+                file.ContentType == "image/jpm" ||
+                // JPEG
+                file.ContentType == "image/jpeg" ||
+                // PNG
+                file.ContentType == "image/png" ||
+                // PSD
+                file.ContentType == "image/vnd.adobe.photoshop" ||
+                file.ContentType == "application/x-photoshop" ||
+                file.ContentType == "application/photoshop" ||
+                file.ContentType == "application/psd" ||
+                file.ContentType == "image/psd" ||
+                // SGI
+                file.ContentType == "image/sgi" ||
+                // TGA
+                file.ContentType == "image/x-targa" ||
+                file.ContentType == "image/x-tga" ||
+                // TIFF
+                file.ContentType == "image/tiff" ||
+                file.ContentType == "image/tiff-fx"
+                //// Last Resort
+                //file.ContentType == "application/octet-stream"
+
+                )
+            {
+                try
+                {
+                    // Delete old File before writing the new one.
+                    try
+                    {
+                        HelpersThing.DeleteFilesForUser(id);
+                    }
+                    catch (Exception err)
+                    {
+
+                    }
+
+                    var path = Path.Combine(
+                                Directory.GetCurrentDirectory(), "images",
+                                id + Path.GetExtension(file.FileName));
+
+                    using (var stream = new FileStream(path, FileMode.OpenOrCreate))
+                    {
+                        await file.CopyToAsync(stream);
+                    }
+
+                }
+                catch (Exception err)
+                {
+                    return StatusCode(500, "Internal Server Error");
+                };
+
+                return Ok();
+            }
+            else
+            {
+                return BadRequest("Content malformated. Server does only accept image/jpeg");

```

Comment:
After the if above this message is not exactly right. obviously, there are checks for other types than "image/jpeg"